src/java.base/share/classes/java/lang/StableValue.java

```diff
@@ -0,0 +1,747 @@
+     * <p>
+     * The returned {@linkplain Supplier supplier} is a caching supplier that records
+     * the value of the provided {@code original} supplier upon being first accessed via
+     * the returned supplier's {@linkplain Supplier#get() get()} method.
+     * <p>
+     * The provided {@code original} supplier is guaranteed to be successfully invoked
+     * at most once even in a multi-threaded environment. Competing threads invoking the
+     * returned supplier's {@linkplain Supplier#get() get()} method when a value is
+     * already under computation will block until a value is computed or an exception is
+     * thrown by the computing thread.
+     * <p>
+     * If the provided {@code original} supplier throws an exception, it is relayed
+     * to the initial caller and no content is recorded.
+     * <p>
+     * If the provided {@code original} supplier recursively calls the returned
+     * supplier, an {@linkplain IllegalStateException} will be thrown.
+     *
+     * @param original supplier used to compute a cached value
+     * @param <T>      the type of results supplied by the returned supplier
+     */
+    static <T> Supplier<T> supplier(Supplier<? extends T> original) {
+        Objects.requireNonNull(original);
+        return StableSupplier.of(original);
+    }
+
+    /**
+     * {@return a new stable {@linkplain IntFunction}}
+     * <p>
+     * The returned function is a caching function that, for each allowed {@code int}
+     * input, records the values of the provided {@code original}
+     * function upon being first accessed via the returned function's
+     * {@linkplain IntFunction#apply(int) apply()} method. If the returned function is
+     * invoked with an input that is not allowed, an {@link IllegalArgumentException}
+     * will be thrown.
+     * <p>
+     * The provided {@code original} function is guaranteed to be successfully invoked
+     * at most once per allowed input, even in a multi-threaded environment. Competing
+     * threads invoking the returned function's
+     * {@linkplain IntFunction#apply(int) apply()} method when a value is already under
+     * computation will block until a value is computed or an exception is thrown by
+     * the computing thread.
+     * <p>
+     * If the provided {@code original} function throws an exception, it is relayed
+     * to the initial caller and no content is recorded.
+     * <p>
+     * If the provided {@code original} function recursively calls the returned
+     * function for the same index, an {@linkplain IllegalStateException} will
+     * be thrown.
+     *
+     * @param size     the size of the allowed inputs in {@code [0, size)}
+     * @param original IntFunction used to compute cached values
+     * @param <R>      the type of results delivered by the returned IntFunction
+     * @throws IllegalArgumentException if the provided {@code size} is negative.
+     */
+    static <R> IntFunction<R> intFunction(int size,
```

Comment:
Does an intFunction of size 0 make sense?

Reply:
it is similar to an empty list but even more useless.